### PR TITLE
Add proxy/iterator to iterate groups in order.

### PR DESCRIPTION
**Pull request by rpavlik**
Simplifies an idiom seen all over the codebase, of iterating
thru groupOrder then using it to look up a group.

### CODE REVIEW

File: src/solvespace.h [hunk 1]
```diff
@@ -485,6 +485,48 @@ class GCodeFileWriter : public VectorFileWriter {
...
+
+    bool operator==(OrderedGroupsIterator const &other) const {
+        if(IsEnd() && other.IsEnd()) {
+            return true;
+        }
+        return sk == other.sk && current == other.current;
+    }
+    bool operator!=(OrderedGroupsIterator const &other) const { return !((*this) == other); }
+
+private:
```
Comment (whitequark) [inline src/solvespace.h]: We don't really use C++ visibility.

File: src/solvespace.h [hunk 2]
```diff
@@ -485,6 +485,48 @@ class GCodeFileWriter : public VectorFileWriter {
...
+    Sketch *sk      = nullptr;
+    hGroup *current = nullptr;
+};
+
+
+class OrderedGroupsProxy {
```
Comment (whitequark) [inline src/solvespace.h]: Is there any way to simplify this? Returning a pair or something. As it is I feel like this patch adds more complexity than it removes.

Comment (whitequark) [reply src/solvespace.h]: I mean, +70 lines -27 lines.

Comment (rpavlik) [reply src/solvespace.h]: I wish there was - but ranges haven't made it into the standard yet. (Maybe c++20?)

No hard feelings, I can close this. Just had it sitting around here.

**Pull request closed without merging**